include/swift/AST/Decl.h

```diff
@@ -462,7 +482,12 @@ class alignas(1 << DeclAlignInBits) Decl {
     /// Initializer stubs can be invoked from Objective-C or through
     /// the Objective-C runtime; there is no way to directly express
     /// an object construction that will invoke a stub.
-    HasStubImplementation : 1
+    HasStubImplementation : 1,
+
+    /// If this constructor is implicit, the implicit constructor kind. If the
+    /// constructor is not implicit, the value is meaningless. Used to provide
+    /// better cross module initializer access diagnostics.
+    ImplicitConstructorKind : NumImplicitConstructorKindBits
```

Comment:
We already have setIsMemberwiseInitializer(), etc. Can you consolidate some of this state?